documentation/general/decouple-vs-and-net-sdk.md

```diff
@@ -0,0 +1,180 @@
+3. **Clear Handling of Divergence**: To make explicit that it is okay, and even expected, that the design time and command line build experiences can differ when the two products are not in sync.
+
+## Terminology
+
+- **msbuild**: refers the .NET Framework based msbuild included with Visual Studio.
+- **dotnet build**: refers to the .NET Core based msbuild included with the .NET SDK.
+- **analyzers**: this document will use analyzers to refer to both analyzers and generators.
+- **torn state**: when the .NET SDK and Visual Studio are not in sync.
+- **.NET SDK project**: a project that uses the .NET SDK project format.
+- **anaylzers**: refer to both analyzers and generators unless otherwise specified.
+
+## Summary
+
+Visual Studio and .NET SDK are separate products but they are intertwined in command line and design time build scenarios as different components are loaded from each product. This table represents how intertwined Roslyn specifically is currently:
+
+| Scenario | Loads Roslyn | Loads Analyzers / Generators |
+| --- | --- | --- |
+| msbuild | From Visual Studio | From .NET SDK |
+| dotnet build | From .NET SDK | From .NET SDK |
+| Visual Studio Design Time | From Visual Studio | From Both |
+
+There are other products that have similar intertwined behavior such as NuGet, MSBuild, Razor, etc ... Roslyn is generally the most impactful and the one we are focusing on in this document as it's path forward can be generalized to the other products.
+
+Generally this mixing of components is fine because Visual Studio will install a .NET SDK that is functionally paired with it. For example 17.10 installs .NET SDK 8.0.3xx, 17.9 installs .NET SDK 8.0.2xx, etc ... In that scenario the components are the same versio and the mixing is largely benign. However when the customer is in a torn state, that is the .NET SDK used is not paired with the Visual Studio version, then compatibility issues can, and **will**, arise. Particularly when the .NET SDK is _newer_ than Visual Studio customers will end up with the following style of error:
+
+> CSC : warning CS9057: The analyzer assembly '..\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk.Razor\source-
+> generators\Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll' references version '4.9.0.0' of the compiler, which is newer
+> than the currently running version '4.8.0.0'.
+
+These bugs slip by because even though a given .NET SDK is supported in many Visual Studio versions, we do not fully validate it's correctness. Specifically consider that when releasing the .NET 8.0.400 SDK it is supported in 17.8-17.11 but only fully validated in 17.11. Likewise when Visual Studio releases servicing fixes it does not attempt to fully validate all the supported .NET SDKs. This work on both sides would be prohibitively expensive. Instead we seek to decouple these experiences so that such testing is not necessary.
```

Comment:
```suggestion
These bugs slip by because even though a given .NET SDK is supported in many Visual Studio versions, we do not fully validate its correctness. Specifically, consider that when releasing the .NET 8.0.400 SDK. It is supported in 17.8-17.11 but only fully validated in 17.11. Likewise, when Visual Studio releases servicing fixes it does not attempt to fully validate all supported .NET SDKs. This work on both sides would be prohibitively expensive. Instead, we seek to decouple these experiences so that such testing is not necessary.
```